### DISCUSSION TITLE
cant debug Vlang code

### DISCUSSION THREAD

**Discussion by lucas8338** [vlang/v#17903] [Questions and Answers] [answered]
i'm trying to debug a vlang code in vscode but my breakpoints in vscode are never being hit. (breakpoints are not working).

i have tried to compile the code using:
`v -cc tcc -keepc -cg main.v`
no success. breakpoints keeps not working.

how should i to configure the vscode to debug vlang?

my launch.json config:
> {
>     "configurations": [
>     {
>         "name": "(gdb) Launch",
>         "type": "cppdbg",
>         "request": "launch",
>         "program": "${workspaceFolder}/${fileBasenameNoExtension}.exe",
>         "args": [],
>         "stopAtEntry": false,
>         "cwd": "${fileDirname}",
>         "environment": [],
>         "externalConsole": false,
>         "MIMode": "gdb",
>         "miDebuggerPath": "C:/mingw64/bin/gdb.exe",
>         "setupCommands": [
>             {
>                 "description": "Enable pretty-printing for gdb",
>                 "text": "-enable-pretty-printing",
>                 "ignoreFailures": true
>             },
>             {
>                 "description": "Set Disassembly Flavor to Intel",
>                 "text": "-gdb-set disassembly-flavor intel",
>                 "ignoreFailures": true
>             }
>         ]
>     }
>     ]
> }

my vscode version:
> Version: 1.76.2 (user setup)
> Commit: ee2b180d582a7f601fa6ecfdad8d9fd269ab1884
> Date: 2023-03-14T17:55:54.936Z
> Electron: 19.1.11
> Chromium: 102.0.5005.196
> Node.js: 16.14.2
> V8: 10.2.154.26-electron.0
> OS: Windows_NT x64 10.0.19042
> Sandboxed: No

my pc:
> OS: Windows 10 Pro 20H2
> CPU: Intel(R) Core(TM)2 Quad CPU    Q9500  @ 2.83GHz   2.83 GHz

**Comment by JalonSolov** [comment]
To debug comfortably, compile with gcc instead of tcc.  tcc doesn't output debug info that the debugger likes.

**Comment by i582** [comment, marked as answer]
Hi, try without the `-c` flag and use GCC to be sure:

```
v -cc gcc -keepc -g main.v
```

`-c` tells V not to create special `#line` directives in C code, and that's the problem.

See https://github.com/vlang/v/blob/master/doc/docs.md#c-backend-binaries-default

**Comment by lucas8338** [comment]
it worked, i thought gcc was being used by default by v, thats why i was using the '-cc tcc' flag.
ussing `v -cc gcc -g main.v` works.

**Reply by JalonSolov** [reply]
tcc is used by default, because it is so much faster than any of the others.  gcc/clang are used if you specifically add `-cc <compiler name>`, or if you add the `-prod` switch.
